src/main/java/com/novi/springboottechiteasy/services/RemoteControllerService.java

```diff
@@ -0,0 +1,147 @@
+
+        } else {
+            throw new RecordNotFoundException("No remote-controller found with id: " + id);
+        }
+    }
+
+    public void deleteRemoteController(Long id) {
+        remoteControllerRepository.deleteById(id);
+    }
+
+    public RemoteControllerDto updateRemoteControllerDetails(Long id, RemoteControllerInputDto updatedRemoteController) {
+        Optional<RemoteController> remote = remoteControllerRepository.findById(id);
+
+        if (remote.isPresent()) {
+
+            RemoteController thisRemote = remote.get();
+            if (updatedRemoteController.getCompatibleWith() != null) {
+                thisRemote.setCompatibleWith(updatedRemoteController.getCompatibleWith());
+            }
+            if (updatedRemoteController.getBatteryType() != null) {
+                thisRemote.setBatteryType(updatedRemoteController.getBatteryType());
+            }
+            if (updatedRemoteController.getName() != null) {
+                thisRemote.setName(updatedRemoteController.getName());
+            }
+            if (updatedRemoteController.getPrice() != null) {
+                thisRemote.setPrice(updatedRemoteController.getPrice());
+            }
+            if (updatedRemoteController.getOriginalStock() != null) {
+                thisRemote.setOriginalStock(updatedRemoteController.getOriginalStock());
+            }
+            if (updatedRemoteController.getTelevisionId() != null) {
+                Television television = televisionRepository.findById(updatedRemoteController.getTelevisionId())
+                        .orElseThrow(() -> new RecordNotFoundException("No television found with id: " + updatedRemoteController.getTelevisionId()));
+
+                thisRemote.setTelevision(television);
+            }
+
+            RemoteController saveRemote = remoteControllerRepository.save(thisRemote);
+
+            return remoteControllerDtoMapper.mapToDto(saveRemote);
+
+        } else {
+            throw new RecordNotFoundException("No remote-controller found with id: " + id);
+        }
+    }
+
+    public RemoteController transferToRemoteControl(RemoteControllerInputDto inputDto) {
+        RemoteController remote = new RemoteController();
+
+        remote.setCompatibleWith(inputDto.getCompatibleWith());
+        remote.setBatteryType(inputDto.getBatteryType());
+        remote.setName(inputDto.getName());
+        remote.setPrice(inputDto.getPrice());
+        remote.setOriginalStock(inputDto.getOriginalStock());
+
+        if (inputDto.getTelevisionId() != null) {
+            TelevisionDto televisionDto = televisionService.getTelevisionById(inputDto.getTelevisionId());
+            if (televisionDto != null) {
+                Television television = new Television();
+                // als er een tv gevonden wordt met deze id, map deze dan naar de Television,
+                // zodat de remote deze television via de setter kan toewijzen
```

Comment:
Dit moet je wel nog even doen, want nu wordt en een lege Television gezet. Je television relatie in RemoteController heeft echter geen cascade, dus dit gaat waarschijnlijk niet werken.